[16.0][IMP] edi_oca: allow to receive empty files

QuocDuong1306 opened the pull request:
Currently when we receive an empty file:
- Receive step doesn't end up in state error_on_receive : that's because we just ignore the content [here](https://github.com/OCA/edi-framework/blob/77184146551ea042190ca7b0a1c290f848dd202f/edi_oca/models/edi_backend.py#L530)
- But the [process step will raise an error](https://github.com/OCA/edi-framework/blob/77184146551ea042190ca7b0a1c290f848dd202f/edi_oca/models/edi_backend.py#L460)
- So this PR goal tries to improve that
- By default, this option is disabled so we still consider that an empty file is an error case

simahawk commented on edi_oca/models/edi_backend.py:
```diff
@@ -579,6 +580,21 @@ def _exchange_receive(self, exchange_record):
             return component.receive()
         raise NotImplementedError()
 
+    def _process_received_content(self, exchange_record, content):
```
IMO this should be done in `validate_data` and you can set the content of the file directly.
I would:
* keep the code as it was before but remove the check on the content value.
* set the value as it is on the record
* call validate data and raise the error there if necessary

simahawk commented on edi_oca/models/edi_backend.py:
```diff
@@ -579,6 +580,21 @@ def _exchange_receive(self, exchange_record):
+        if len(content) == 0:
+            if not exchange_record.type_id.allow_empty_files_on_receive:
+                raise ValueError(
+                    "File is Empty and Empty Files are not allowed for this exchange type"
```
```suggestion
                    _("Empty files are not allowed for this exchange type")
```

simahawk commented:
`(disabled by default)` should be moved to the commit description together w/ the PR description :wink:

Commit:
[IMP] edi_oca: allow to receive empty files

Currently when we receive an empty file:

- Receive step doesn't end up in state error_on_receive : that's because we just ignore the content here
- But the process step will raise an error
- So this commit goal tries to improve that
- By default, this option is disabled so we still consider that an empty file is an error case

QuocDuong1306 commented:
Hi @simahawk , I updated the PR

simahawk commented:
Ooops... I hit the merge btn by error :sweat:
